### PR TITLE
refactor orchestration config for apollo.

### PR DESCRIPTION
**Pull request by dongzl**
For #3185 .

Changes proposed in this pull request:
- fixes orchestration config listener for nacos and apollo.
- refactor orchestration config for apollo.
- fixes nacos config key name.

### CODE REVIEW

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstance.java [hunk 1]
```diff
@@ -17,177 +17,93 @@
...
 import java.util.Properties;
-import java.util.Set;
 
 /**
- * Registry center for Apollo.
+ * Config center for Apollo.
  *
  * @author dongzonglei
  */
-@Slf4j
```
Comment (kimmking) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstance.java]: why remove slf4j?

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloConfigWrapperTest.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+import java.util.concurrent.TimeUnit;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+public final class ApolloConfigWrapperTest {
+    
+    static {
+        SLF4JBridgeHandler.removeHandlersForRootLogger();
```
Comment (kimmking) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloConfigWrapperTest.java]: why remove?

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstance.java (hunk 1 above)
Comment (dongzl) [reply sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstance.java]: There are not log print, so I remove this Annotation.

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloConfigWrapperTest.java (hunk 2 above)
Comment (dongzl) [reply sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloConfigWrapperTest.java]: For close java.util.log print, only use logback.

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstance.java [hunk 3]
```diff
@@ -17,177 +17,93 @@
...
-        return apolloConfig.getProperty(convertKey(key), "");
-    }
-    
-    private String convertKey(final String shardingSphereKey) {
-        return shardingSphereKey.replace(SHARDING_SPHERE_KEY_SEPARATOR, APOLLO_KEY_SEPARATOR).substring(1);
-    }
-    
-    private String deConvertKey(final String apolloKey) {
-        return new StringBuilder(SHARDING_SPHERE_KEY_ROOT).append(apolloKey.replace(APOLLO_KEY_SEPARATOR, SHARDING_SPHERE_KEY_SEPARATOR)).toString();
+        String value = configWrapper.getProperty(ConfigKeyUtils.path2Key(key));
```
Comment (wgy8283335) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstance.java]: String value = configWrapper.getProperty(ConfigKeyUtils.path2Key(key));
or
String newKey = ConfigKeyUtils.path2Key(key);
String value = configWrapper.getProperty(newKey);
Which one is better, accroding to clean code?

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstance.java [hunk 4]
```diff
@@ -17,177 +17,93 @@
...
+        return Strings.isNullOrEmpty(value) ? openApiWrapper.getValue(ConfigKeyUtils.path2Key(key)) : value;
     }
     
     @Override
     public List<String> getChildrenKeys(final String key) {
```
Comment (wgy8283335) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstance.java]: I think maybe we could remove getChildrenKeys from config center interface.

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstance.java [hunk 5]
```diff
@@ -17,177 +17,93 @@
...
                     DataChangedEvent.ChangedType changedType = getChangedType(change.getChangeType());
-                    if (DataChangedEvent.ChangedType.IGNORED != changedType) {
-                        dataChangedEventListener.onChange(new DataChangedEvent(deConvertKey(key), change.getNewValue(), changedType));
+                    if (DataChangedEvent.ChangedType.IGNORED == changedType) {
+                        continue;
+                    }
+                    if (caches.get(changeKey) == null) {
+                        continue;
                     }
+                    caches.get(changeKey).onChange(new DataChangedEvent(ConfigKeyUtils.key2Path(changeKey), change.getNewValue(), changedType));
```
Comment (wgy8283335) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstance.java]: Is this line too long?

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloConfigWrapper.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+ */
+public final class ApolloConfigWrapper {
+    
+    private static final String APOLLO_KEY_APP_ID = "app.id";
+    
+    private static final String APOLLO_KEY_ENV = "env";
+    
+    private Config apolloConfig;
+    
+    public ApolloConfigWrapper(final InstanceConfiguration config, final Properties properties) {
```
Comment (wgy8283335) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloConfigWrapper.java]: Does this function need a document comment?

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloOpenApiWrapper.java [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    
+    private String appId;
+    
+    private String env;
+    
+    private String clusterName;
+    
+    private String administrator;
+    
+    public ApolloOpenApiWrapper(final InstanceConfiguration config, final Properties properties) {
```
Comment (wgy8283335) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloOpenApiWrapper.java]: Does this function need a document comment?

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloOpenApiWrapper.java [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+        return itemDTO.getValue();
+    }
+    
+    /**
+     * Persist config.
+     * 
+     * @param key key
+     * @param value value
+     */
+    public void persist(final String key, final String value) {
```
Comment (wgy8283335) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloOpenApiWrapper.java]: Does this need some code to make sure that updateKey could rollback when publishNameSpace fail? For example: TCC.

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstance.java (hunk 4 above)
Comment (wgy8283335) [reply sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstance.java]: And close() method also could be discussed.

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloConfigWrapperTest.java (hunk 2 above)
Comment (wgy8283335) [reply sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloConfigWrapperTest.java]: Dose apollo default use the logback to print log?

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-nacos/pom.xml [hunk 9]
```diff
@@ -39,7 +35,7 @@
         <dependency>
             <groupId>com.alibaba.nacos</groupId>
             <artifactId>nacos-client</artifactId>
-            <version>${nacos-client.verison}</version>
+            <version>${nacos.client.verison}</version>
             <scope>provided</scope>
```
Comment (wgy8283335) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-nacos/pom.xml]: Why is provided?

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-api/src/main/java/org/apache/shardingsphere/orchestration/center/util/ConfigKeyUtils.java [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+ */
+
+package org.apache.shardingsphere.orchestration.center.util;
+
+/**
+ * Config key utils.
+ *
+ * @author dongzonglei
+ */
+public final class ConfigKeyUtils {
```
Comment (wgy8283335) [inline sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-api/src/main/java/org/apache/shardingsphere/orchestration/center/util/ConfigKeyUtils.java]: Whether should provide public utility method in interface module?

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloConfigWrapperTest.java (hunk 2 above)
Comment (dongzl) [reply sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/test/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloConfigWrapperTest.java]: Apollo use java.util.log, it is in conflict with shardingsphere, so it should be remove.

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloConfigWrapper.java (hunk 6 above)
Comment (dongzl) [reply sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloConfigWrapper.java]: This is a construct method,  construct method doesn't need comment on ShardingSphere.

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloOpenApiWrapper.java (hunk 7 above)
Comment (dongzl) [reply sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloOpenApiWrapper.java]: The same to above.

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-api/src/main/java/org/apache/shardingsphere/orchestration/center/util/ConfigKeyUtils.java (hunk 10 above)
Comment (dongzl) [reply sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-api/src/main/java/org/apache/shardingsphere/orchestration/center/util/ConfigKeyUtils.java]: This is only use for orchestration ConfigCenter module right now, I think we will remove it to a common module in the future If need.

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloOpenApiWrapper.java (hunk 7 above)
Comment (dongzl) [reply sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloOpenApiWrapper.java]: The same to above.

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-nacos/pom.xml (hunk 9 above)
Comment (dongzl) [reply sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-nacos/pom.xml]: nacos is not a default implemention, it provides for user selection, so it is provided.

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloOpenApiWrapper.java (hunk 8 above)
Comment (dongzl) [reply sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloOpenApiWrapper.java]: I think we can implement the function and optimize it in the future. For Apollo, if publishNameSpace fail, user can manual management in the Apollo's portal.

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstance.java (hunk 3 above)
Comment (dongzl) [reply sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstance.java]: I don't think there's any difference between them, I think thery are OK.

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstance.java (hunk 4 above)
Comment (dongzl) [reply sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstance.java]: The etcd does not integrate now, I think if it doesn't need getChildrenKeys, we can remove it.

close() method should be retain, but it is not need for all framework.

File: sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstance.java (hunk 5 above)
Comment (dongzl) [reply sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstance.java]: ShardingSphere's checkstyle does not warn, so it is in one line.

File: sharding-proxy/sharding-proxy-bootstrap/src/main/resources/conf/server.yaml [hunk 11]
```diff
@@ -29,11 +29,6 @@
 #    namespace: orchestration
 #    props:
 #      overwrite: false
-#      retry-interval-milliseconds: 10
```
Comment (kimmking) [inline sharding-proxy/sharding-proxy-bootstrap/src/main/resources/conf/server.yaml]: why remove?

**Comment by kimmking** [comment]
@dongzl  you can close this pr and submit it to master directly.

**Comment by dongzl** [comment]
Close this pr, and submit a new pr to master branch directly.

**Pull request closed without merging**